Incorrect floating-point truncation in ImageService

Specific image dimensions and scale may prevent image from rendering with an exacption

```
Caused by: java.awt.image.RasterFormatException: x lies outside raster
at java.desktop/sun.awt.image.IntegerInterleavedRaster.createWritableChild(IntegerInterleavedRaster.java:458)
at java.desktop/java.awt.image.BufferedImage.getSubimage(BufferedImage.java:1203)
at com.enonic.xp.core.impl.image.effect.ScaledFunction.apply(ScaledFunction.java:25)
at com.enonic.xp.core.impl.image.ImageServiceImpl.createImage(ImageServiceImpl.java:237)
at com.enonic.xp.core.impl.image.ImageServiceImpl.writeImage(ImageServiceImpl.java:101)
at com.enonic.xp.core.impl.image.ImageServiceImpl.lambda$readImage$1(ImageServiceImpl.java:90)
at 
```

--- a/modules/core/core-image/src/main/java/com/enonic/xp/core/impl/image/effect/ScaleCalculator.java
+++ b/modules/core/core-image/src/main/java/com/enonic/xp/core/impl/image/effect/ScaleCalculator.java
@@ -46,11 +46,11 @@ public interface ScaleCalculator
             if ( heightScale < widthScale )
             {
                 newWidth = width;
-                newHeight = (int) ( sourceHeight * widthScale );
+                newHeight = Math.max( height, (int) ( sourceHeight * widthScale ) );
             }
             else
             {
-                newWidth = (int) ( sourceWidth * heightScale );
+                newWidth = Math.max( width, (int) ( sourceWidth * heightScale ) );
                 newHeight = height;
             }
 

--- a/modules/core/core-image/src/test/java/com/enonic/xp/core/impl/image/effect/BlockScaleTest.java
+++ b/modules/core/core-image/src/test/java/com/enonic/xp/core/impl/image/effect/BlockScaleTest.java
@@ -73,6 +73,26 @@ class BlockScaleTest
     }
 
     @Test
+    public void same_aspect_ratio_upscale()
+    {
+        final BufferedImage original = ImageHelper.createImage( 800, 600, false );
+
+        BufferedImage scaled = newScaleFunctions().block( FocalPoint.DEFAULT, 804, 603 ).apply( original );
+        assertEquals( 804, scaled.getWidth() );
+        assertEquals( 603, scaled.getHeight() );
+    }
+
+    @Test
+    public void same_aspect_ratio_downscale()
+    {
+        final BufferedImage original = ImageHelper.createImage( 1600, 1200, false );
+
+        BufferedImage scaled = newScaleFunctions().block( FocalPoint.DEFAULT, 804, 603 ).apply( original );
+        assertEquals( 804, scaled.getWidth() );
+        assertEquals( 603, scaled.getHeight() );
+    }
+
+    @Test
     void upscale_square()
     {
         final BufferedImage original = ImageHelper.createImage( 400, 300, false );

--- a/modules/core/core-image/src/test/java/com/enonic/xp/core/impl/image/effect/ScaleCalculatorTest.java
+++ b/modules/core/core-image/src/test/java/com/enonic/xp/core/impl/image/effect/ScaleCalculatorTest.java
@@ -52,6 +52,15 @@ class ScaleCalculatorTest
     }
 
     @Test
+    void block_same_aspect_ratio_floating_point_truncation()
+    {
+        final ScaleCalculator.Values values = ScaleCalculator.block( 804, 603, 0.5, 0.5 ).calc( 800, 600 );
+
+        assertThat( values ).extracting( v -> v.newWidth, v -> v.newHeight, v -> v.widthOffset, v -> v.heightOffset, v -> v.viewWidth,
+                                         v -> v.viewHeight ).containsExactly( 804, 603, 0, 0, 804, 603 );
+    }
+
+    @Test
     void block_focal_lower_right()
     {
         final ScaleCalculator.Values values = ScaleCalculator.block( 10, 10, 1, 1 ).calc( 200, 100 );

